spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/SdkTracerProviderCustomizer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.tracing;
+
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+
+/**
+ * Callback interface that can be used to customize auto-configured
+ * {@link SdkTracerProviderBuilder}.
+ *
+ * @author Yanming Zhou
+ * @since 3.1.0
+ */
+@FunctionalInterface
+public interface SdkTracerProviderCustomizer {
```

Comment:
I think this should be called `SdkTracerProviderBuilderCustomizer`. We can take care of that as part of merging it.

Reply:
Actually I named it that at first, after some thoughts, I think `SdkTracerProviderCustomizer` is identical to `SdkTracerProviderBuilderCustomizer`, because `SdkTracerProvider` cannot constructed without `SdkTracerProviderBuilder`, and the only purpose of `SdkTracerProviderBuilder` is building `SdkTracerProvider`.

Reply:
I think we should aim for consistency with other similar interfaces that already exist. We have many that are named `…BuilderCustomizer` because the class that they are customised is a `…Builder`. The same applies here.